src/App.tsx

```diff
@@ -1,96 +1,63 @@
-import React from 'react';
+import React, { useState } from 'react';
+
 import './App.css';
-import { getNumbers } from './utils';
+import { getItems } from './services/utils';
+import { Pagination } from './components/Pagination';
+import { ItemList } from './components/ItemList';
+import { PageSizeSelectList } from './components/PageSizeSelectList';
+import { defaultPaginVal } from './utils';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42)
-  .map(n => `Item ${n}`);
+const PAGE_SIZE_OPTIONS = [3, 5, 10, 20];
 
 export const App: React.FC = () => {
+  const [paginationOption, setPaginationOption] = useState({
+    total: defaultPaginVal.total,
+    perPage: defaultPaginVal.perPage,
+    currentPage: defaultPaginVal.currentPage,
+  });
+
+  const currentPageHandler = (value: number) => setPaginationOption({
+    ...paginationOption,
+    currentPage: value,
+  });
```

Comment:
If we want to work with the actual state, it should be received through a callback, and not applied directly.